sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -335,6 +358,19 @@ trait Row extends Serializable {
   def getAs[T](fieldName: String): T = getAs[T](fieldIndex(fieldName))
 
   /**
+   * Returns the value of a given fieldName as a scala.util.Try object.
+   */
+  def attempt[T](fieldName: String): Try[T] = Try(getAs[T](fieldIndex(fieldName)))
```

Comment:
This part of the code isn't my department, so take this with a grain of salt. I'm not sure what to make of `attempt` as an API method. (I know from the discussion what it's for.) I'd expect to see getters, or maybe getters that return an `Option`. There's already a method that gets a value or `null`, which can be made into an `Option` by the caller. Right? or else I miss what else it's for. I assume this is kind of mimicking the JDBC API which would return nulls for nullable values.